test/acme_server/nonce_test.exs

```diff
@@ -0,0 +1,61 @@
+defmodule AcmeServer.NonceTest do
+  use ExUnit.Case, async: true
+  use ExUnitProperties
+
+  test "new is unique" do
+    first_nonce = AcmeServer.Nonce.new()
+    nonce = AcmeServer.Nonce.new()
+    assert nonce != first_nonce
+  end
+
+  test "verify! created nonce" do
+    nonce = AcmeServer.Nonce.new()
+    assert :ok == AcmeServer.Nonce.verify!(nonce)
+  end
+
+  test "verify! created nonce ONLY once" do
+    nonce = AcmeServer.Nonce.new()
+    assert :ok == AcmeServer.Nonce.verify!(nonce)
+
+    assert_raise MatchError, fn ->
+      AcmeServer.Nonce.verify!(nonce)
+    end
+  end
+
+  test "verify! unknown nonce throw error" do
+    assert_raise MatchError, fn ->
+      AcmeServer.Nonce.verify!(-1)
+    end
+  end
```

Comment:
I kept the unit tests, as I find them useful to demonstrate the usage of a function (versus it's properties).